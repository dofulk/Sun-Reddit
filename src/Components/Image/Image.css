

.Image {
  border-radius: 3px;
  max-height: 70vh;
  max-width: 100%;
  object-fit: scale-down;
}