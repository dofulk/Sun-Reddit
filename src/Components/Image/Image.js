import React, { Component } from 'react';
import './Image.css'

class Image extends Component {


  render() {
    const { src } = this.props;
    return (
      <div className="Image-Container">
        <img className="Image" onLoad={this.onImgLoad} src={src} alt="Sorry"/>
      </div>
    )
  }
}

export default Image