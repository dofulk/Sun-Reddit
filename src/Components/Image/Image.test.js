import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image';
import Cat from '../../cateyes.jpg';

it('renders without props', () => {
  const tree = renderer.create(<Image />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders with image', () => {
  const tree = renderer.create(<Image src={Cat}/>).toJSON();
  expect(tree).toMatchSnapshot();
})