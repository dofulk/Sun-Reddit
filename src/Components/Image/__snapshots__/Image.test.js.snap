// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with image 1`] = `
<div
  className="Image-Container"
>
  <img
    alt="Sorry"
    className="Image"
    src="cateyes.jpg"
  />
</div>
`;

exports[`renders without props 1`] = `
<div
  className="Image-Container"
>
  <img
    alt="Sorry"
    className="Image"
  />
</div>
`;
